src/librustc/middle/ty.rs

```diff
@@ -3128,7 +3128,6 @@ pub fn type_is_scalar(ty: Ty) -> bool {
       ty_bool | ty_char | ty_int(_) | ty_float(_) | ty_uint(_) |
       ty_infer(IntVar(_)) | ty_infer(FloatVar(_)) |
       ty_bare_fn(..) | ty_ptr(_) => true,
-      ty_tup(ref tys) if tys.is_empty() => true,
```

Comment:
It seems to me that this might have non-local consequence, beyond just changing these error messages slightly, maybe just changing the internals of the cast checking would be more reliable?